libpod/options.go

```diff
@@ -548,14 +553,78 @@ func WithRuntimeFlags(runtimeFlags []string) RuntimeOption {
 
 // Container Creation Options
 
+// WithCpuSetCpusCtr computes and sets the Cpus linux resource string which determines the amount of cores, from those available,  we are allowed to execute on
+func WithCpuSetCpusCtr(res string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return define.ErrRuntimeFinalized
+		}
+
+		wholeNum := ""
+		for j := 0; j < len(res); j++ {
+			temp := res[j]
+			tempS := string(temp)
+			if len(wholeNum) == 0 || len(wholeNum) == 1 {
+				wholeNum = tempS
+			}
+			if val, err := strconv.Atoi(wholeNum); err == nil {
+				if int(runtime.NumCPU()) < val {
+					max := fmt.Sprint(runtime.NumCPU())
+					totalNew := "0-" + max
+					res = totalNew
+				}
+			}
+			if j != len(res)-1 {
+				if (string(res[j+1]) != "-") && (string(res[j+1]) != ",") && wholeNum != "-" && wholeNum != "," {
+					wholeNum += string(res[j+1])
+				}
+			}
+		}
```

Comment:
@cdoern Let's chat tomorrow about your goal here and how to improve the code performance.

Reply:
Okay, thank you @jwhonce! I just implemented a slightly better way using regex, I basically want to check if the cpuset-cpus string contains a CPU core value greater than the total number of cores available. If it does, I reset the value to `0-availableCores`. I left the other code in a comment just in case we wan to look at it.